app/controllers/v0/sessions_controller.rb

```diff
@@ -7,6 +7,9 @@ class SessionsController < ApplicationController
     include Accountable
 
     skip_before_action :authenticate, only: %i[new logout saml_callback saml_logout_callback]
+    before_action do
+      @relay_state = RelayState.new(relay_enum: params[:success_relay], url: params[:RelayState])
```

Comment:
I don't really like doing this in a before action like this. Can we move this to a method? And make sure the before_action is only invoked for the actions that actually require it?

`before_action :initialize_relay_state, only: %i[new saml_callback saml_logout_callback]`

alternately and this might be even better, just create a method relay_state that memoizes.

```ruby
private

def relay_state
   @relay_state ||= RelayState.new(relay_enum: params[:success_relay], url: params[:RelayState])
end
```

Reply:
I like the memoized method a lot.  Will do 👍